modules/vcf_metal_device/main.tf

```diff
@@ -3,12 +3,13 @@ resource "equinix_metal_device" "esx" {
   timeouts {
     create = "60m"
   }
-  hostname         = var.esxi_hostname
-  project_id       = var.metal_project_id
-  metro            = var.metal_metro
-  plan             = var.metal_device_plan
-  operating_system = var.esxi_version_slug
-  billing_cycle    = var.metal_billing_cycle
+  hostname                = var.esxi_hostname
+  project_id              = var.metal_project_id
+  metro                   = var.metal_metro
+  plan                    = var.metal_device_plan
+  hardware_reservation_id = "next-available"
```

Comment:
We may want the wait for full deprovision option to accompany this and an optional hardware_reservation_id, defaulting to "next-available".  This will let users override the hardware reservation in the root module.

Reply:
Unrelated, but in the `firstboot_shell_cmd = <<EOT` lines, you can use `<<-EOT` and then indent the code to make it a little easier to read.

Reply:
I've rewired this to have the reservation Id in the block that defines the four ESXi servers.